pulser/simulation/simulation.py

```diff
@@ -641,26 +641,24 @@ def make_xy_term(masked: bool = False) -> qutip.Qobj:
             includes a 1/hbar factor.
             """
             # Calculate the total number of good, unmasked qubits
-            effective_size = (
-                self._size
-                - len(self._seq._slm_mask_targets) * int(masked)
-                - sum(self._bad_atoms.values())
-            )
-            if effective_size < 2:
-                return 0 * self.build_operator([("I", "global")])
+            if masked:
+                effective_size = self._size - sum(self._bad_atoms.values())
+                for q in self._seq._slm_mask_targets:
+                    if not self._bad_atoms[q]:
+                        effective_size -= 1
+                if effective_size < 2:
+                    return 0 * self.build_operator([("I", "global")])
 
             xy = cast(qutip.Qobj, 0)
             # Get every pair without duplicates
             for q1, q2 in itertools.combinations(self._qdict.keys(), r=2):
                 if (
                     self._bad_atoms[q1]
                     or self._bad_atoms[q2]
-                    or (
-                        masked
-                        and (
-                            q1 in self._seq._slm_mask_targets
-                            or q2 in self._seq._slm_mask_targets
-                        )
+                    or masked
+                    and (
+                        q1 in self._seq._slm_mask_targets
+                        or q2 in self._seq._slm_mask_targets
```

Comment:
Why did you remove the parameters? Now it will do this every time it is `masked`

Reply:
sorry I'm not sure what happened there. It should be fixed now